replica/database.hh

```diff
@@ -193,7 +193,7 @@ private:
     seal_immediate_fn_type _seal_immediate_fn;
     std::function<schema_ptr()> _current_schema;
     replica::dirty_memory_manager* _dirty_memory_manager;
-    memtable_table_shared_data _table_shared_data;
+    memtable_table_shared_data& _table_shared_data;
```

Comment:
```c++
// Statistics that need to be shared across all memtables for a single table
struct memtable_table_shared_data {
    logalloc::allocating_section read_section;
    logalloc::allocating_section allocating_section;
};
```

So copying _table_shared_data was wrong.

Looks like it was born wrong in 2a76065e3d3b2e9bc8cc18d2436bec0aa8d98d14.

Reply:
It was also copying the two separate members even before that. But before that, in the vnode world, there was one memtable_list per table, so the memtable_list members were in fact table wide. The transition to tablets and many memtable_lists per table forgot to account for this.